static/src/javascripts-legacy/projects/commercial/modules/user-features.js

```diff
@@ -93,7 +93,11 @@ define([
         expiryDate.setDate(expiryDate.getDate() + 1);
         cookies.addCookie(USER_FEATURES_EXPIRY_COOKIE, expiryDate.getTime().toString());
         cookies.addCookie(PAYING_MEMBER_COOKIE, !JsonResponse.adblockMessage);
-        cookies.addCookie(AD_FREE_USER_COOKIE, JsonResponse.adFree);
+        if(JsonResponse.adFree) {
+            cookies.addCookie(AD_FREE_USER_COOKIE, JsonResponse.adFree, 1);
+        } else {
+            cookies.removeCookie(AD_FREE_USER_COOKIE);
```

Comment:
Even though you are removing the cookies, I'd still make sure that your fastly vcl has only two scenarios:
1. brand new cache key for ad-free tier
2. identical key for all other circumstances
ie. don't just latch the cookie value into the cache key. Instead, conditionally add it only if the cookie value is positive.